src/main/java/org/numenta/nupic/network/Layer.java

```diff
@@ -1909,12 +1919,36 @@ private void clearSubscriberObserverLists() {
      * @return
      */
     NamedTuple makeClassifiers(MultiEncoder encoder) {
+        Map inferredFields = (Map<String, Class<? extends Classifier>>) params.get(KEY.INFERRED_FIELDS);
+        if(inferredFields == null || inferredFields.entrySet().size() == 0) {
```

Comment:
Please replace lines 1920, 1921, and 1922 with the following in order to suppress "raw generic warning":

@SuppressWarnings("unchecked")
NamedTuple makeClassifiers(MultiEncoder encoder) {
        Map<String, Class<? extends Classifier>> inferredFields = (Map<String, Class<? extends Classifier>>) params.get(KEY.INFERRED_FIELDS);